src/common/collect_data/script.sh

```diff
@@ -16,34 +16,43 @@ for task_id in label_projection dimensionality_reduction denoising; do
 
   mkdir -p $out_dir
 
+  # generate task info
+  viash run src/common/get_task_info/config.vsh.yaml -- \
+    --input "src" \
+    --query $task_id \
+    --output "$out_dir/task_info.json"
+
   # generate method info
-  viash run src/common/get_method_info/config.vsh.yaml -p native -- \
-    --input "src/$task_id" \
+  viash run src/common/get_method_info/config.vsh.yaml -- \
+    --input "src" \
+    --query $task_id \
     --output "$out_dir/temp_method_info.json"
-  viash run src/common/check_migration_status/config.vsh.yaml -p native -- \
+  viash run src/common/check_migration_status/config.vsh.yaml -- \
     --git_sha "openproblems_git.json" \
     --comp_info "$out_dir/temp_method_info.json" \
     --output "$out_dir/method_info.json"
   rm "$out_dir/temp_method_info.json"
 
   # generate metric info
-  viash run src/common/get_metric_info/config.vsh.yaml -p native -- \
-    --input "src/$task_id" \
+  viash run src/common/get_metric_info/config.vsh.yaml -- \
+    --input "src" \
+    --query $task_id \
     --output "$out_dir/temp_metric_info.json"
-  viash run src/common/check_migration_status/config.vsh.yaml -p native -- \
+  viash run src/common/check_migration_status/config.vsh.yaml -- \
     --git_sha "openproblems_git.json" \
     --comp_info "$out_dir/temp_metric_info.json" \
     --output "$out_dir/metric_info.json"
   rm "$out_dir/temp_metric_info.json"
   
   # generate results
-  viash run src/common/get_results/config.vsh.yaml -p native -- \
+  viash run src/common/get_results/config.vsh.yaml -- \
     --input_scores "resources/$task_id/benchmarks/openproblems_v1/combined.extract_scores.output.tsv" \
     --input_execution "resources/$task_id/benchmarks/openproblems_v1/nextflow_log.tsv" \
     --output "$out_dir/results.json"
   
   # generate api info
-  viash run src/common/get_api_info/config.vsh.yaml -p native -- \
-    --input "src/$task_id" \
+  viash run src/common/get_api_info/config.vsh.yaml -- \
+    --input "src" \
+    --query $task_id \
```

Comment:
How about changing the interface for all of these components (`get_api_info`, `get_task_info`, `get_method_info`) such that the input directory is the full `openproblems-v2` repo and you can choose which task_id to process. For example, `get_task_info --input ../openproblems-v2 --task_id label_projection --output foo.json`?

Reply:
You might even be able to specify an API for it so that the parameters are all the same ;)